www/assets/js/components/SearchPage.js

```diff
@@ -76,21 +76,23 @@ export default function SearchPage() {
 
       setSearchFacets(new Map(Object.entries(newResults.aggregations ?? {})))
 
-      setSearchResults(
+      setSearchResults(results =>
```

Comment:
That's interesting, I wish they would highlight the need for this in the docs for handling previous state

Reply:
yeah...they discuss it in the docs but don't really clarify how helpful it is for avoiding race conditions for these kinds of things. It also means if you're passing the state update function in to a `useCallback` function you can avoid also passing the current value as a dependency, which I think would in a lot of cases drastically cut down on the number of unnecessary re-allocations of that function (which will increase memory usage and pressure on the GC and whatnot).